pandas/tests/indexes/test_multi.py

```diff
@@ -3307,3 +3307,40 @@ def test_duplicate_multiindex_labels(self):
         with pytest.raises(ValueError):
             ind.set_levels([['A', 'B', 'A', 'A', 'B'], [2, 1, 3, -2, 5]],
                            inplace=True)
+
+    def test_use_enum_in_multiindex(self):
+        # GH 21298
+        # Allow use of Enums as one of the factors in a MultiIndex.
+
+        # See https://github.com/pandas-dev/pandas/blob/
+        # 0c65c57a279e755ab7093db925d1e580f9878dae/pandas/util/testing.py#L793-L799.
+        # We cannot simply use tm.assert_index_equal, as the "expected" index
+        # cannot actually be built yet.
+        from pandas.core.algorithms import take_1d
+
+        def _get_ilevel_values(index, level):
```

Comment:
what is all this for?

simply construct the expected index directly

Reply:
The thought here is that the expected index _cannot_ be constructed directly, due to the exact bug I'm trying to fix. Thus instead I create an expected "level", and verify that it matches the relevant level of `df`'s index. This is done by replicating a sub-step of `tm.assert_index_equal`.